### PR TITLE
Dev nick

### PR DESCRIPTION
**Pull request by NickIs211**

### CODE REVIEW

File: app/controllers/tutoring_session_controller.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def index
 
       start_week += 1.day # Offset by a day due to how time parses out when converted from cookie
     else
-      start_week = Date.today.beginning_of_week(start_day = :sunday).to_datetime
+      start_week = Date.today.beginning_of_week.to_datetime
```
Comment (baileyjrob) [inline app/controllers/tutoring_session_controller.rb:22]: I assume it starts on monday by default?

File: app/models/role.rb [hunk 2]
```diff
@@ -4,4 +4,12 @@
 class Role < ApplicationRecord
   has_and_belongs_to_many :users
   validates :role_name, presence: true
+
+  def self.get_admin_role
+    Role.where('role_name like ?', 'Admin').first
```
Comment (baileyjrob) [inline app/models/role.rb]: Clever, I had completely forgotten that models could take functions

File: app/models/tutoring_session.rb [hunk 3]
```diff
@@ -20,24 +20,22 @@ def top_offset
   end
 
   def end_of_semester_datetime
-    ret = ('May 12th, ' + scheduled_datetime.year.to_s).to_datetime
-    if scheduled_datetime > ret
-      ret = ('December 17th, ' + scheduled_datetime.year.to_s).to_datetime
-    end
+    ret = "May 12th, #{scheduled_datetime.year}".to_datetime
+    ret = "December 17th, #{scheduled_datetime.year}".to_datetime if scheduled_datetime > ret
```
Comment (baileyjrob) [inline app/models/tutoring_session.rb:24]: Excellent use of Rubocop

File: db/seeds.rb [hunk 4]
```diff
@@ -8,13 +8,10 @@
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
 
-User.create!(id: 0, first_name: 'Admin', last_name: 'User', email: 'admin@tamu.edu', password: 'T3st!!a')
-Role.create([{ role_name: 'Admin' }, { role_name: 'Tutor' }, { role_name: 'Student' }, { role_name: 'Spartan Tutor' }])
-User.create!(first_name: 'John', last_name: 'Doe', email: 'john@tamu.edu', password: 'T3st!!b')
-user2 = User.create!(first_name: 'Jane', last_name: 'Doe', email: 'jane@tamu.edu', password: 'T3st!!c')
-user3 = User.create!(first_name: 'Jeff', last_name: 'Doe', email: 'jeff@tamu.edu', password: 'T3st!!d')
+admin_role = Role.create(role_name: 'Admin');
+Role.create(role_name: 'tutor');
```
Comment (baileyjrob) [inline db/seeds.rb]: Shouldn't either both be capitalized or both be lowercase? This seems like it's asking for errors. Also, why is it assigning to a variable? Where is that going?
```suggestion
admin_role = Role.create(role_name: 'Admin');
Role.create(role_name: 'Tutor');
```

File: spec/feature/tutoring_session_scheduler_spec.rb [hunk 5]
```diff
@@ -1,211 +1,219 @@
...
+  let(:tutor) { User.where(first_name: 'Tutor', last_name: 'User').first }
+  let(:scheduled_datetime) { '26 May 2021 08:00:00 +0000'.to_datetime }
+  let(:beginning_of_week) { Date.today.beginning_of_week }
+
+  after { Timecop.return }
+
+  before do
+    Timecop.freeze(frozen_time)
+    User.create(first_name: 'Admin', last_name: 'User', password: 'T3st!!a',
+                email: 'admin@tamu.edu')
```
Comment (baileyjrob) [inline spec/feature/tutoring_session_scheduler_spec.rb:15]: Couldn't lines 14/15 replace the part in the brackets on line 6? It'd be more intuitive, and it would result in fewer database interactions
```suggestion
  let!(:tutor) { User.create(first_name: 'Admin', last_name: 'User', password: 'T3st!!a',
                                       email: 'admin@tamu.edu') }
  let(:scheduled_datetime) { '26 May 2021 08:00:00 +0000'.to_datetime }
  let(:beginning_of_week) { Date.today.beginning_of_week }

  after { Timecop.return }

  before do
    Timecop.freeze(frozen_time)
```

File: spec/feature/user_login_spec.rb [hunk 6]
```diff
@@ -19,9 +19,6 @@
 
   describe 'SHOW' do
     it 'shows user' do
-      admin = User.where(first_name: 'Admin').first
-      visit("users/#{admin.id}")
-      expect(page).to have_content('Admin User')
```
Comment (baileyjrob) [inline spec/feature/user_login_spec.rb]: Why did you remove this? Doesn't this just remove the test entirely?

**Review by baileyjrob — CHANGES_REQUESTED** [review]

File: app/controllers/tutoring_session_controller.rb (hunk 1 above)
Comment (NickIs211) [reply app/controllers/tutoring_session_controller.rb:22]: Yes

File: spec/feature/tutoring_session_scheduler_spec.rb [hunk 7]
```diff
@@ -1,211 +1,219 @@
...
+    User.create(first_name: 'Tutor', last_name: 'User', password: 'T3st!!a',
+                email: 'tutor@tamu.edu')
```
Comment (baileyjrob) [inline spec/feature/tutoring_session_scheduler_spec.rb]: Couldn't lines 16/17 replace the part in the brackets on line 6? It'd be more intuitive, and it would result in fewer database interactions
```suggestion
  let!(:tutor) { User.create(first_name: 'Tutor', last_name: 'User', password: 'T3st!!a',
                email: 'tutor@tamu.edu') }
  let(:scheduled_datetime) { '26 May 2021 08:00:00 +0000'.to_datetime }
  let(:beginning_of_week) { Date.today.beginning_of_week }

  after { Timecop.return }

  before do
    Timecop.freeze(frozen_time)
    User.create(first_name: 'Admin', last_name: 'User', password: 'T3st!!a',
                                       email: 'admin@tamu.edu')
```

**Review by baileyjrob — CHANGES_REQUESTED** [review]

File: db/seeds.rb (hunk 4 above)
Comment (NickIs211) [reply db/seeds.rb]: Fixed in latest commit

File: spec/feature/user_login_spec.rb (hunk 6 above)
Comment (NickIs211) [reply spec/feature/user_login_spec.rb]: Unintended commit, meant to revert the file after I saw my tests passing.

File: spec/feature/tutoring_session_scheduler_spec.rb (hunk 7 above)
Comment (NickIs211) [reply spec/feature/tutoring_session_scheduler_spec.rb]: I'm not sure if that destroys the tutor object after every test like what happens in the before section. If it's the same code, looks good to me.

File: spec/feature/user_login_spec.rb (hunk 6 above)
Comment (baileyjrob) [reply spec/feature/user_login_spec.rb]: Does this mean I'm good to just put the lines back in?

Comment (NickIs211) [reply spec/feature/user_login_spec.rb]: Yeah, this change can be reverted
